homeassistant/components/dynalite/bridge.py

```diff
@@ -1,21 +1,16 @@
 """Code to handle a Dynalite bridge."""
 
-from typing import TYPE_CHECKING, Any, Callable, Dict, List
+from typing import Any, Callable, Dict, List, Optional
 
 from dynalite_devices_lib.dynalite_devices import DynaliteDevices
 
 from homeassistant.const import CONF_HOST
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.dispatcher import async_dispatcher_send
 
-from .const import CONF_ALL, ENTITY_PLATFORMS, LOGGER
+from .const import ENTITY_PLATFORMS, LOGGER
 from .convert_config import convert_config
 
-if TYPE_CHECKING:  # pragma: no cover
```

Comment:
This needs to remain there or else the type checking won't work. (you can test it with `mypy <path to file>`)

Reply:
that is exactly the weird behavior i had. when i moved it from DynaliteBaseDevice to Optional, flake8 didn't let me commit because it said it is unused. When I removed it, flake8 worked and mypy had no problem (see checks below). Weird.
Perhaps I can remove the Optional[] here? Since we don't define the flag '--no-implicit-optional", if it is defined with a default value of NULL, mypy understands its Optional by default?

Reply:
in any case, i figured out that i don't need "if TYPE_CHECKING" since there is no circular dependency, so while I still don't understnad why it happened, the problem is no longer relevant

Reply:
Oh I missed that you import it now from the package directly. ok 👍